src/LangOpts.zig

```diff
@@ -0,0 +1,49 @@
+const std = @import("std");
+
+const LangOpts = @This();
+
+const Standard = enum {
+    /// ISO C 1990
+    c89,
+    /// ISO C 1990 with GNU extensions
+    gnu89,
+    /// ISO C 1999
+    c99,
+    /// ISO C 1999 with GNU extensions
+    gnu99,
+    /// ISO C 2011
+    c11,
+    /// ISO C 2011 with GNU extensions
+    gnu11,
+    /// ISO C 2017
+    c17,
+    /// ISO C 2017 with GNU extensions
+    gnu17,
+    /// Working Draft for ISO C2x
+    c2x,
+    /// Working Draft for ISO C2x with GNU extensions
+    gnu2x,
+
+    const Invalid = error{InvalidStandard};
+
+    fn fromString(name: []const u8) Invalid!Standard {
+        return std.meta.stringToEnum(Standard, name) orelse error.InvalidStandard;
```

Comment:
This should use a `ComptimeStringMap` since there are some common aliases.

Here's the list given by clang when specifying an invalid version:
```
error: invalid value 'c87' in '-std=c87'
note: use 'c89', 'c90', or 'iso9899:1990' for 'ISO C 1990' standard
note: use 'iso9899:199409' for 'ISO C 1990 with amendment 1' standard
note: use 'gnu89' or 'gnu90' for 'ISO C 1990 with GNU extensions' standard
note: use 'c99' or 'iso9899:1999' for 'ISO C 1999' standard
note: use 'gnu99' for 'ISO C 1999 with GNU extensions' standard
note: use 'c11' or 'iso9899:2011' for 'ISO C 2011' standard
note: use 'gnu11' for 'ISO C 2011 with GNU extensions' standard
note: use 'c17', 'iso9899:2017', 'c18', or 'iso9899:2018' for 'ISO C 2017' standard
note: use 'gnu17' or 'gnu18' for 'ISO C 2017 with GNU extensions' standard
note: use 'c2x' for 'Working Draft for ISO C2x' standard
note: use 'gnu2x' for 'Working Draft for ISO C2x with GNU extensions' standard
```